lib/compress/zstd_fast.c

```diff
@@ -51,10 +51,12 @@ size_t ZSTD_compressBlock_fast_generic(
     U32* const hashTable = ms->hashTable;
     U32 const hlog = cParams->hashLog;
     /* support stepSize of 0 */
-    U32 const stepSize = cParams->targetLength + !(cParams->targetLength);
+    size_t const stepSize = cParams->targetLength + !(cParams->targetLength) + 1;
```

Comment:
I had this as `2 * (targetLength + !targetLength)`, but I think that `+1` is a better choice.

This will make negative compression levels slightly stronger and slower, but the user can set any negative level, so that should be fine.